doc/specs/#16599 - Quick Fix.md

```diff
@@ -0,0 +1,56 @@
+---
+author: Carlos Zamora @carlos-zamora
+created on: 2024-04-01
+last updated: 2024-04-05
+issue id: #16599
+---
+# Quick Fix
+## Solution Design
+Quick Fix will be a new UI surface that allows the user to interact with the terminal and leverage the context of a specific command being executed. The UI is discussed in the [UI/UX Design section](#ui/ux-design).
+
+The UI itself will live in the TerminalControl layer, but it can be populated by both the TerminalControl and the TerminalApp layer (similar to the context menu). The suggestions will be appended as top-level items, but a nested design can be revisited as more sources are added to populate the Quick Fix menu. Only one Quick Fix menu will be available at a time as it only appears on the latest prompt if an error occurred; this is in alignment with VS Code's design.
```

Comment:
> as it only appears on the latest prompt

how are we placing that / knowing where a prompt is? I forget - do we emit the `osc 9001` at the start of the error message, or the end?

Reply:
> how are we placing that / knowing where a prompt is?

So, actually, implementation-wise, I'm not really checking for when the next prompt is ready:
![image](https://github.com/microsoft/terminal/assets/11050425/f70d3ffc-8f1e-4d52-b1d3-6947777bfe1f)

It's been working for me so far (** knock on wood **). Any suggestions on a better way to do it?
 
> do we emit the `osc 9001` at the start of the error message, or the end?

We emit OSC 9001 at the start of the error message.